src/main/java/org/mockito/internal/junit/StubbingArgMismatches.java

```diff
@@ -0,0 +1,49 @@
+package org.mockito.internal.junit;
+
+import org.mockito.internal.util.MockitoLogger;
+import org.mockito.invocation.Invocation;
+
+import java.util.*;
+
+/**
+ * Contains stubbing arg mismatches, knows how to format them
+ */
+class StubbingArgMismatches {
+
+    final Map<Invocation, Set<Invocation>> mismatches = new LinkedHashMap<Invocation, Set<Invocation>>();
+
+    public void add(Invocation invocation, Invocation stubbing) {
+        Set<Invocation> matchingInvocations = mismatches.get(stubbing);
+        if (matchingInvocations == null) {
+            matchingInvocations = new LinkedHashSet<Invocation>();
+            mismatches.put(stubbing, matchingInvocations);
+        }
+        matchingInvocations.add(invocation);
+    }
+
+    public void format(String testName, MockitoLogger logger) {
+        if (mismatches.isEmpty()) {
+            return;
+        }
+
+        StubbingHint hint = new StubbingHint(testName);
+        //TODO 544 it would be nice to make the String look good if x goes multiple digits (padding)
```

Comment:
Use `java.lang.String#format(java.lang.String, java.lang.Object...)`, something like :

```java
String.format("%4d. %s %s %n", x++, ". Unused... ", m.getKey().getLocation())
```

`4` is hard coded here, but that would work up to 9999 hint. It's still possible to compute that value if hard coded value don't work.